pkg/scheduler/framework/plugins/helper/preemption.go

```diff
@@ -0,0 +1,430 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helper
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+
+	v1 "k8s.io/api/core/v1"
+	policy "k8s.io/api/policy/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	policylisters "k8s.io/client-go/listers/policy/v1"
+	corev1helpers "k8s.io/component-helpers/scheduling/corev1"
+	"k8s.io/klog/v2"
+	extenderv1 "k8s.io/kube-scheduler/extender/v1"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	"k8s.io/kubernetes/pkg/scheduler/metrics"
+	"k8s.io/kubernetes/pkg/scheduler/util"
+)
+
+// PreemptionEvaluator is expected to be implemented by different preemption plugins as all those member
+// methods could have different behavior as to the default preemption.
+type PreemptionEvaluator interface {
```

Comment:
The architecture we want to achieve is something like we did for https://github.com/kubernetes/kubernetes/tree/master/pkg/scheduler/framework/plugins/noderesources

That is, plugins would create some sort of "preemption scorer" object for which you can configure some functions.
I'm not quite liking the dependency of this package on the framework types.